reframe/frontend/executors/__init__.py

```diff
@@ -261,14 +264,22 @@ def setup(self, *args, **kwargs):
         self._notify_listeners('on_task_setup')
 
     def compile(self):
-        self._safe_call(self.check.compile)
+        try:
+            self._safe_call(self.check.compile)
+            self._notify_listeners('on_task_compile')
+        except JobQOSMaxSubmitJobPerUserLimitError as e:
+            raise e
```

Comment:
It is not nice that we treat `JobQOSMaxSubmitJobPerUserLimitError` differently here and in the `run` method below, but I couldn't find a better way without changing the structure of policies and how we poll for compiled jobs.